src/language/concurrency.md

```diff
@@ -144,80 +168,106 @@ void main() [!async!] {
+system (OS). Once `readAsString()` returns a value, Dart code execution resumes.
 
-![Flowchart-like figure showing app code executing from start to exit, waiting for native I/O in between](/assets/img/language/concurrency/basics-await.png)
+![Flowchart-like figure showing app code executing from start to exit, waiting 
+for native I/O in between](/assets/img/language/concurrency/basics-await.png)
 
-If you'd like to learn more about using `async`, `await`, and futures,
-visit the [asynchronous programming codelab][].
+### Stream syntax
 
-[asynchronous programming codelab]: /codelabs/async-await
+Dart also supports asynchronous code in the form of streams. Streams
+provide values in the future and repeatedly over time. A promise to provide a
+series of `int` values over time has the type `Stream<int>`.
+
+This example will return a new `int` value every second.
+
+<?code-excerpt "lib/stream_syntax.dart"?>
+```dart
+Stream<int> stream = Stream.periodic(const Duration(seconds: 1), (i) => i * i);
+```
+
+#### await-for and yield
 
+Await-for is a type of for loop that executes each subsequent iteration of the
+loop as new values are provided. In other words, it’s used to “loop over”
+streams. In this example, a new value will be returned from the function
+`sumStream` as new values are emitted from the stream that’s provided as an
+argument. The `yield` keyword is used rather than `return` in functions that return
+streams of values. 
 
-## How isolates work
+<?code-excerpt "lib/await_for_syntax.dart"?>
+```dart
+Stream<int> sumStream(Stream<int> stream) async* {
+  var sum = 0;
+  await for (final value in stream) {
+    yield sum += value;
+  }
+}
+```
 
-Most modern devices have multi-core CPUs.
-To take advantage of multiple cores,
-developers sometimes use shared-memory threads running concurrently.
-However, shared-state concurrency is
-[error prone](https://en.wikipedia.org/wiki/Race_condition#In_software) and
-can lead to complicated code.
+If you'd like to learn more about using `async`, `await`, `Stream`s and 
+`Future`s,  visit the [asynchronous programming codelab][].
 
-Instead of threads, all Dart code runs inside of isolates.
-Each isolate has its own memory heap,
-ensuring that none of the state in an isolate is accessible from
-any other isolate.
-No shared state between isolates means concurrency complexities like 
+[asynchronous programming codelab]: /codelabs/async-await
+
+## Isolates
+
+Dart also supports concurrency via isolates. Most modern devices have multi-core
+CPUs. To take advantage of multiple cores, developers sometimes use
+shared-memory threads running concurrently. However, shared-state concurrency is
+[error prone](https://en.wikipedia.org/wiki/Race_condition#In_software) and can
+lead to complicated code.
+
+Instead of threads, all Dart code runs inside of isolates. Each isolate has its
+own memory heap, ensuring that none of the state in an isolate is accessible
```

Comment:
I know this is pre-existing, but since we're here, let's avoid the term "heap" since they can share the same heap. It sounds like "has its own global fields" is the most accurate, but "has its own memory" is probably fine too.